content/en/blog/2023/traffic-for-ambient-and-sidecar/index.md

```diff
@@ -0,0 +1,107 @@
+    - the waypoint proxy is scheduled to Node C
+- foo namespace enables sidecar injection (foo namespace contains label: `istio-injection: enabled`)
+- bar-1 namespace enables ambient mode (bar-1 namespace contains label: `istio.io/dataplane-mode=ambient`)
+
+According to above description, the deployment and network traffic path are showing below:
+
+{{< image width="100%"
+    link="sidecar-to-ambient.png"
+    caption="sleep to httpbin and helloworld"
+    >}}
+
+### [ sleep -> sidecar ] -> ztunnel -> httpbin network traffic path
+
+Based on the above deployment, the first part is showing on top half of the diagram: the network traffic path of sleep pod in Sidecar mode to httpbin pod in Ambient mode.
+
+**(1) (2) (3) (4)** the sleep container sends a request to httpbin, the request is intercepted by iptables rules and directed to port 15001 on the sleep sidecar. Then, the sidecar handles the L7 load balancing and routing rules by following configurations distributed by the control plane. Next, the request traffic is forwarded to a specific httpbin pod in node B, identified by its concrete ip address.
+
+**(5) (6)**  After the request is sent to the  device pair (`veth httpbin <-> eth0 inside httpbin pod`),  the request would be intercepted and forwarded to the `istioin` device in the node B where httpbin pod is running by following its iptables rules and route rules. The `istioin` device in the node B is a `[geneve](https://www.rfc-editor.org/rfc/rfc8926.html)` tunnel, and the other side of this tunnel is pair to `pistioin` device which is inside the ztunnel pod of node B.
+
+**(7) (8)** After the request enters the `pistioin` device of the ztunnel pod, the request is intercepted to port 15008 of the ztunnel pod by the iptables rules of the ztunnel pod in node B.
+
+**(9)** The traffic getting into the port 15008 would be considered as a inbound request, then ztunnel will forward the request to the httpbin pod in the same node B.
+
+### [ sleep-> sidecar ] -> waypoint -> ztunnel -> helloworld httpbin network traffic path
+
+Comparing with the first part of the diagram, it's clear that the waypoint proxy path is added in path **"[ sleep -> sidecar ] -> ztunnel -> httpbin"**, and the extra configuration of waypoint proxy for helloworld service causes this.
+
+What's the background story?
+
+The Istiod control plane has all information of service and configuration in the service mesh. When helloworld is configured with waypoint proxy, the EDS configuration of helloworld service received by sidecar of sleep pod will be changed to type of `envoy_internal_address`, this causes that the request traffic going through the internal of envoy will be forwarded to port 15008 of waypoint proxy in node C via `[HBONE](https://docs.google.com/document/d/1Ofqtxqzk-c_wn0EgAXjaJXDHB9KhDuLe-W3YGG67Y8g/edit)` protocol.
+
+Similar with sidecar, waypoint proxy is based on envoy and it will forward the request to helloworld pod based on envoy L7 routing strategy.
```

Comment:
```suggestion
Similar to sidecar, waypoint proxy is based on envoy and it will forward the request to the helloworld pod based on envoy L7 routing strategy.
```

Reply:
already done!